code/model/modelanimation.cpp

```diff
@@ -12,12 +12,20 @@ extern float flFrametime;
 
 namespace animation {
 
-	flag_def_list_new<animation::Animation_Flags> Animation_flags[] = {
-		{ "auto reverse",				animation::Animation_Flags::Auto_Reverse,						true, false },
-		{ "reset at completion",		animation::Animation_Flags::Reset_at_completion,		        true, false },
-		{ "loop",						animation::Animation_Flags::Loop,						        true, false },
-		{ "random starting phase",		animation::Animation_Flags::Random_starting_phase,				true, false },
-		{ "pause on reverse",			animation::Animation_Flags::Pause_on_reverse,					true, false },
+	special_flag_def_list_new<animation::Animation_Flags, ModelAnimation&> Animation_flags[] = {
+		{ "auto reverse",				animation::Animation_Flags::Auto_Reverse,						true },
+		{ "reset at completion",		animation::Animation_Flags::Reset_at_completion,		        true },
+		{ "loop",						animation::Animation_Flags::Loop,						        true },
+		{ "random starting phase",		animation::Animation_Flags::Random_starting_phase,				true },
+		{ "pause on reverse",			animation::Animation_Flags::Pause_on_reverse,					true },
+		{ "seamless with startup",		animation::Animation_Flags::Seamless_with_startup,				true, [](const SCP_string& from, ModelAnimation& anim) {
+			anim.m_flagData.loopsFrom = static_cast<float>(std::atof(from.c_str()));
+			anim.m_flags.set(animation::Animation_Flags::Loop);
+			anim.m_flags.set(animation::Animation_Flags::Seamless_with_startup);
+
+			//While this flag implies Reset_at_completion semantically, it's implemented in a conflicting manner. Hence, make sure that seamless will take priority, and force-disable reset at completion if set
+			anim.m_flags.set(animation::Animation_Flags::Reset_at_completion, false);
```

Comment:
This works if the "reset at completion" flag comes first, but if it comes later, then this will not prevent both flags from being set.  So some sort of check should happen after all flags are parsed.

Reply:
This works due to the way the parsing stuff does things.
First, all non-special flags are parsed and set. Afterwards, all special or unparsed flags are checked and their functor is run.
So as long as you only modify non-special flags, this kind of flag modification should be safe

Reply:
Ah, ok.